build/tools/ExtractRuleDescFromJson/RuleDescExtractor.cs

```diff
@@ -87,26 +88,61 @@ private void PrepareDestinationDirectory()
         }
     }
 
-    private static IEnumerable<Rule> LoadRules(string file)
+    private static IEnumerable<PluginRule> LoadRules(string file)
     {
+        var options = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
+
         var json = File.ReadAllText(file);
-        return JsonSerializer.Deserialize<Rule[]>(json)
-            ?? Array.Empty<Rule>();
+        return JsonSerializer.Deserialize<PluginRule[]>(json, options)
+            ?? Array.Empty<PluginRule>();
     }
 
-    private void ProcessRule(Rule rule)
+    private void ProcessRule(PluginRule pluginRule)
     {
         try
         {
-            var xml = EnsureHtmlIsXml(rule.Description);
-            SaveRuleFile(rule, xml);
+            var descAsxml = EnsureHtmlIsXml(pluginRule.Description);
```

Comment:
Previously we were just saving a text file.
Now we need to create a new json file with more info in it, in a form that can be deserialized by the product code.